test/UrlParserTest.cpp

```diff
@@ -0,0 +1,71 @@
+// Copyright 2024, University of Freiburg,
+// Chair of Algorithms and Data Structures.
+// Author: Julian Mundhahs (mundhahj@tf.uni-freiburg.de)
+
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+
+#include "util/GTestHelpers.h"
+#include "util/http/HttpUtils.h"
+#include "util/http/UrlParser.h"
+
+using namespace ad_utility;
+
+TEST(UrlParserTest, paramsToMap) {
+  auto parseParams =
+      [](const string& queryString) -> HashMap<std::string, std::string> {
+    boost::urls::url_view url(queryString);
+    return UrlParser::paramsToMap(url.params());
+  };
+  auto HashMapEq = [](const HashMap<std::string, std::string>& hashMap)
+      -> testing::Matcher<HashMap<std::string, std::string>> {
+    return testing::ContainerEq(hashMap);
+  };
+
+  EXPECT_THROW(parseParams("?foo=a&foo=b"), std::runtime_error);
+  // Parameter with key "" is set twice.
+  EXPECT_THROW(parseParams("?&"), std::runtime_error);
```

Comment:
Those are errors that are propoagated to the user, hence the messages matter.
Thus you should use `AD_EXPECT_THROW_WITH_MESSAGE` here (not that the "wrong" runtime error is thrown). It suffices to test a subset of the query.